tests/base.py

```diff
@@ -798,6 +813,14 @@ def getPull(self, owner, project, number):
         }
         return data
 
+    def getPullBySha(self, sha):
+        prs = list(set([p for p in self.pull_requests if sha == p.head_sha]))
+        if len(prs) > 1:
+            raise Exception('Multiple pulls found with head sha: %s' % sha)
```

Comment:
Yeah, this is possible, it would mean that there are two identical PRs (or rather two PRs with differently-named but identical branches). Definitely a strange and unlikely corner case, but since we don't know which is correct I think an exception is correct.